plugins/setoid_ring/newring.ml

```diff
@@ -153,9 +153,11 @@ let decl_constant na univs c =
   let open Constr in
   let vars = CVars.universes_of_constr c in
   let univs = UState.restrict_universe_context univs vars in
-  let univs = Monomorphic_entry univs in
+  let () = Declare.declare_universe_context false univs in
+  let types = (Typeops.infer (Global.env ()) c).uj_type in
```

Comment:
I guess this could be a Retyping call but the econstr boilerplate makes it easier with typeops

Reply:
Well, technically it doesn't generate the same type in general. This version is backwards-compatible, although maybe a little slower. That said, this shouldn't matter at all here, this is only called at a ring definition time, i.e. virtually never.